presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/ParquetPredicateUtils.java

```diff
@@ -90,65 +90,66 @@ public static boolean isStatisticsOverflow(Type type, ParquetIntegerStatistics p
                 continue;
             }
 
-            Optional<RichColumnDescriptor> descriptor = getDescriptor(fileSchema, requestedSchema, ImmutableList.of(columnHandle.getName()));
-            if (descriptor.isPresent()) {
-                predicate.put(descriptor.get(), entry.getValue());
+            RichColumnDescriptor descriptor = descriptorsByPath.get(ImmutableList.of(columnHandle.getName()));
+            if (descriptor != null) {
```

Comment:
Descriptor can be null if this is a partition key, and doesn't exist in parquet schema. Another case is when column exists in Hive table but doesn't exist in parquet schema (this may happen due to schema evolution, old parquet files may not contain newly added columns). Columns that are not exist in parquet schema can't help in predicate pushdown, so we just skip them.